components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1317,58 +1305,24 @@ void AdsServiceImpl::OnGetAdsHistory(OnGetAdsHistoryCallback callback,
   // Build the list structure required by the webUI
   int uuid = 0;
   base::ListValue list;
-  for (const auto& item : ads_history.items) {
-    base::DictionaryValue ad_content_dictionary;
-    ad_content_dictionary.SetKey("adType",
-                                 base::Value(item.ad_content.type.value()));
-    ad_content_dictionary.SetKey("uuid", base::Value(item.ad_content.uuid));
-    ad_content_dictionary.SetKey(
-        "creativeInstanceId",
-        base::Value(item.ad_content.creative_instance_id));
-    ad_content_dictionary.SetKey("creativeSetId",
-                                 base::Value(item.ad_content.creative_set_id));
-    ad_content_dictionary.SetKey("campaignId",
-                                 base::Value(item.ad_content.campaign_id));
-    ad_content_dictionary.SetKey("brand", base::Value(item.ad_content.brand));
-    ad_content_dictionary.SetKey("brandInfo",
-                                 base::Value(item.ad_content.brand_info));
-    ad_content_dictionary.SetKey(
-        "brandDisplayUrl", base::Value(item.ad_content.brand_display_url));
-    ad_content_dictionary.SetKey("brandUrl",
-                                 base::Value(item.ad_content.brand_url));
-    ad_content_dictionary.SetKey(
-        "likeAction",
-        base::Value(static_cast<int>(item.ad_content.like_action_type)));
-    ad_content_dictionary.SetKey(
-        "adAction",
-        base::Value(std::string(item.ad_content.confirmation_type)));
-    ad_content_dictionary.SetKey("savedAd",
-                                 base::Value(item.ad_content.is_saved));
-    ad_content_dictionary.SetKey("flaggedAd",
-                                 base::Value(item.ad_content.is_flagged));
-
-    base::DictionaryValue category_content_dictionary;
-    category_content_dictionary.SetKey(
-        "category", base::Value(item.category_content.category));
-    category_content_dictionary.SetKey(
-        "optAction",
-        base::Value(static_cast<int>(item.category_content.opt_action_type)));
 
+  for (const auto& item : ads_history.items) {
     base::DictionaryValue ad_history_dictionary;
+    base::DictionaryValue ad_content_dictionary = item.ad_content.ToValue();
     ad_history_dictionary.SetPath("adContent",
                                   std::move(ad_content_dictionary));
+    base::DictionaryValue category_content_dictionary =
+        item.category_content.ToValue();
     ad_history_dictionary.SetPath("categoryContent",
                                   std::move(category_content_dictionary));
+    base::ListValue ad_history_list;
+    ad_history_list.Append(std::move(ad_history_dictionary));
 
     base::DictionaryValue dictionary;
-
     dictionary.SetKey("uuid", base::Value(std::to_string(uuid++)));
-    auto time = base::Time::FromDoubleT(item.timestamp);
-    auto js_time = time.ToJsTime();
+    const base::Time time = base::Time::FromDoubleT(item.timestamp);
+    const double js_time = time.ToJsTime();
```

Comment:
Seems like the current recommendation is to use `ToJsTimeIgnoringNull`.

Reply:
Fixed